docs/use_cases/fine_grained_authorization/intro.md

```diff
@@ -0,0 +1,180 @@
+are satisfied. The language of valid rules depends on a specific FGA implementation. In this tutorial we
+consider two most common types of rules:
+
+1. Rules with one prerequisite:
+  ```
+  relationship1(object1, object2) and condition(object1, object2) -> relationship2(object1, object2)
+  ```
+  where `condition` is an arbitrary predicate over atributes of `object1` and `object2`.  This rule states
+  that if there exists relationship `relationship1` between `object1` and `object2`, which satisfy `condition`,
+  then `object1` and `object2` are also related by `relationship2`.
+
+2. Rules with two prerequisites:
+  ```
+  relationship1(object1, object2) and relationship2(object2, object3) and condition(object1, object3) -> relationship3(object1, object3)
+  ```
+
+More complex rules, e.g., rules with more than two prerequisites or rules with arbitrary combinations of `and` and `or`
+operators, can be decomposed into these two basic types. Some FGA implementations allow negative prerequisites of the
+form `not relationship1(object1, object2)`. Such rules can be expressed in Feldera, but we leave them out for simplicity.
+
+#### Example
+
+We define the following derived relationships in our file manager example:
+- `group-can-read(group, file)` - `group` is allowed to read `file`.
+- `group-can-write(group, file)` - `group` is allowed to write `file`.
+- `user-can-read(user, file)` - `user` is allowed to read `file`.
+- `user-can-write(user, file)` - `group` is allowed to write `file`.
+
+These relationships are governed by the following rules:
+* **Rule 1:** `editor(group, file) -> group-can-write(group, file)` - if a group is an editor of a file, it can write this file.
+* **Rule 2:** `group-can-write(group, file1) and parent(file1, file2) -> group-can-write(group, file2)` - if a group can write
+  a file, then it can write any of its children.
+* **Rule 3:** `viewer(group, file) -> group-can-read(group, file)` - if a group is a viewer of a file, then it can read this file.
+* **Rule 4:** `group-can-write(group, file) -> group-can-read(group, file)` - the write permission to a file implies the read
+  permission to the same file.
+* **Rule 5:** `group-can-read(group, file1) and parent(file1, file2) -> group-can-read(group, file2)` - if a group can read a file,
+  then it can read any of its children.
+* **Rule 6:** `member(user, group) and group-can-write(group, file) and (not user.is_banned) -> user-can-write(user, file)` - if a user is a member
+  of a group that can write a file and the user is not banned, then the user can write the file.
+* **Rule 7:** `member(user, group) and group-can-read(group, file) and (not user.is_banned) -> user-can-read(user, file)` - if a user is a member
+  of a group that can read a file and the user is not banned, then the user can read the file.
+
+Given these rules, we can derive the following access permissions to file `f1` from the example object graph above:
```

Comment:
this graph can be awfully dense. I wonder whether you compute all of it, or only compute what you need on demand